### PR TITLE
used redux

### PR DESCRIPTION
**Pull request by VaidikNissangi9**
fixed bugs and added redux behaviour

### CODE REVIEW

File: src/actions/actions.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import axios from 'axios'
+
+export const getCenter='getCenter'
```
Comment (iamanvesh) [inline src/actions/actions.js]: constants are usually written in all caps and snake-case.
```suggestion
export const GET_CENTER='GET_CENTER'
```

File: src/actions/actions.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+export const getLocations='getLocations'
+export const reverseLocations='reverseLocations'
+
+export function fetchLocations() {
```
Comment (iamanvesh) [inline src/actions/actions.js]: add documentation to all the functions/actions in [JSDoc](https://devhints.io/jsdoc)

File: src/components/addLocation.js [hunk 3]
```diff
@@ -1,79 +1,37 @@
...
-        ...coordinates,
-        ...event.lngLat
-      })
-    })
-  }
   function handleAddLocation() {
-    axios.post("http://localhost:3001/locations", coordinates, {
+    let center={lng:props.center[0],lat:props.center[1]}
+
+    axios.post("http://localhost:3001/locations",center , {
```
Comment (iamanvesh) [inline src/components/addLocation.js]: Am I right in assuming that this is a WIP and will be moved out to redux action eventually?

File: src/components/locations.js [hunk 4]
```diff
@@ -13,73 +14,101 @@ const styles = {
...
-        currentLocation: {},
-        sort: false,
-        check:false
-    }
-    componentDidMount() {
+const locations = (props) => {
+    // console.log(props)
+    const [check, setCheck] = useState(false)
+    useEffect(() => {
         axios.get('http://localhost:3001/locations', {
```
Comment (iamanvesh) [inline src/components/locations.js]: same here.

File: src/components/publicMap.js [hunk 5]
```diff
@@ -27,23 +35,36 @@ const MapboxGLMap = (props) => {
       mapboxgl: mapboxgl
     }));
     setMap(map)
+
```
Comment (iamanvesh) [inline src/components/publicMap.js]: you need to remove the map when the component unmounts
```suggestion
return () => map.remove();
```

**Review by iamanvesh — CHANGES_REQUESTED** [review]
Please add propTypes, documentation to all the components / functions.

File: src/components/addLocation.js (hunk 3 above)
Comment (VaidikNissangi9) [reply src/components/addLocation.js]: Yes, I'm working on it.

File: src/actions/locationsActions.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+ * Fetches locations 
+ * @param {*} dispatch 
+ */
+export function fetchLocations(dispatch) {
+  axios.get('http://localhost:3001/locations', {
+    headers: {
+      "Content-Type": "application/json",
+      "Authorization": "fake-access-token"
```
Comment (iamanvesh) [inline src/actions/locationsActions.js]: you cannot hard code the access token, you need to use the one coming from the backend.

File: src/components/auth.js [hunk 7]
```diff
@@ -1,20 +1,20 @@
...
-    ls.set('isAuthenticated', true)
-    urlPush();
-  }
+/**
+ * on login moves to respective page
+ * @param {*} urlPush 
+ */
+export function login(urlPush) {
+  ls.set('isAuthenticated', true)
+  urlPush();
```
Comment (iamanvesh) [inline src/components/auth.js]: you need to check if the parameter is a function before calling it

**Review by iamanvesh — COMMENTED** [review]
LGTM but few minor issues that need to be addressed

* Function documentation must be clearer, i.e, describe what the function is doing clearly with data types of the argument & return type.
* Have consistent naming conventions for actions, some actions have payload and some others have username/token etc. All the actions should be of the format `{type, payload}`.

**Review by iamanvesh — APPROVED** [review]

**Pull request merged**